includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1033,6 +1033,18 @@ private function process_payment_with_payment_method( int $order_id ) {
 
 			$this->validate_selected_payment_method_type( $payment_information, $order->get_billing_country() );
 
+			if ( $this->is_order_payment_locked( $order ) ) {
+				// If the request is already being processed, return an error.
+				return [
+					'result'   => 'failure',
+					'redirect' => '',
+					'message'  => __( 'Your payment is already being processed. Please wait.', 'woocommerce-gateway-stripe' ),
+				];
+			}
+
+			// Lock the order before we create and confirm the payment/setup intents to prevent Stripe sending the success webhook before this request is completed.
+			$this->lock_order_payment( $order );
+
```

Comment:
I like how this is readable. Slight concern that it adds an extra "check" step, redundant because checking for an existing lock is already performed inside `lock_order_payment( $order )`. I think we want locking to be ideally with minimal operations as possible, to reduce the surface for race conditions.

Maybe something like this? 
```
// Attempt to acquire lock, bail if already locked
$is_already_locked = $this->lock_order_payment( $order );
if ( $is_already_locked ) {
   return [...];
}
```

What do you think?